OpenCVExample/OpenCVExample/MainPage.xaml.cpp

```diff
@@ -0,0 +1,146 @@
+﻿//
+// MainPage.xaml.cpp
+// Implementation of the MainPage class.
+//
+
+#include "pch.h"
+#include "MainPage.xaml.h"
+#include <opencv2\imgproc\types_c.h>
+#include <opencv2\imgcodecs\imgcodecs.hpp>
+#include <opencv2\core\core.hpp>
+#include <opencv2\imgproc\imgproc.hpp>
+#include <opencv2\objdetect.hpp>
+#include <Robuffer.h>
+
+using namespace OpenCVExample;
+
+using namespace Platform;
+using namespace Windows::Foundation;
+using namespace Windows::Foundation::Collections;
+using namespace Windows::UI::Xaml;
+using namespace Windows::UI::Xaml::Controls;
+using namespace Windows::UI::Xaml::Controls::Primitives;
+using namespace Windows::UI::Xaml::Data;
+using namespace Windows::UI::Xaml::Input;
+using namespace Windows::UI::Xaml::Media;
+using namespace Windows::UI::Xaml::Navigation;
+
+using namespace Windows::UI::Xaml::Media::Imaging;
+using namespace Windows::Storage::Streams;
+using namespace Microsoft::WRL;
+
+
+// The Blank Page item template is documented at http://go.microsoft.com/fwlink/?LinkId=402352&clcid=0x409
+
+const cv::String face_cascade_name = "Assets/haarcascade_frontalface_alt.xml";
+const cv::String body_cascade_name = "Assets/haarcascade_fullbody.xml";
+
+MainPage::MainPage()
+{
+	InitializeComponent();
+}
+
+void  OpenCVExample::MainPage::UpdateImage(const cv::Mat& image)
+{
+	// Create the WriteableBitmap
+	WriteableBitmap^ bitmap = ref new WriteableBitmap(image.cols, image.rows);
+
+	// Get access to the pixels
+	IBuffer^ buffer = bitmap->PixelBuffer;
+	unsigned char* dstPixels = nullptr;
+
+	// Obtain IBufferByteAccess
+	ComPtr<IBufferByteAccess> pBufferByteAccess;
+	ComPtr<IInspectable> pBuffer((IInspectable*)buffer);
+	pBuffer.As(&pBufferByteAccess);
```

Comment:
ComPtr::As can fail too, right?  Do our samples assume success or demonstrate error handling (fwiw, for the Windows UWP samples, they advised us against including error checking)